docs/architecture/adr-043-nft-module.md

```diff
@@ -35,341 +36,224 @@ The current design is based on the work done by [IRISnet team](https://github.co
 
 We will create a module `x/nft`, which contains the following functionality:
 
-- Store and transfer NFTs utilizing `x/bank`.
-- Mint and burn NFTs.
-- Use `x/authz` to implement ERC721-style authorization.
+- Store NFTs and track their ownership.
+- Expose `Keeper` interface for composing modules to mint and burn NFTs.
+- Expose external `Message` interface for users to transfer ownership of their NFTs.
 - Query NFTs and their supply information.
 
 ### Types
 
-#### Metadata
+#### Genre
 
-We define a `Metadata` model for `NFT Type`, which is comparable to an ERC721 contract on Ethereum, under which a collection of `NFT`s can be created and managed.
+We define a model for NFT **Genre**, which is comparable to an ERC721 Contract on Ethereum, under which a collection of NFTs can be created and managed.
 
 ```protobuf
-message Metadata {
-  string type          = 1; // Required, unique key, alphanumeric
-  string name          = 2;
-  string symbol        = 3;
-  string description   = 4;
-  bool mint_restricted = 5;
-  bool edit_restricted = 6;
+message Genre {
+  string id              = 1;
+  string name            = 2;
+  string symbol          = 3;
+  string description     = 4;
+  string uri             = 5;
+  bool mint_restricted   = 10;
+  bool update_restricted = 11;
```

Comment:
are these necessary here? might want to hold off on the encumbrances for a separate ADR?

Reply:
Agreed. The idea is to do a sophisticated encumbrance design rather than just stopgaps right? Also as I understand it the encumbrances should be declarative and queryable right?

Reply:
> are these necessary here? might want to hold off on the encumbrances for a separate ADR?

Yes, it makes sense to have the app-level modules to enforce such encumbrances in ways that best suit their needs.